docker-build.sh

```diff
@@ -0,0 +1,355 @@
+BUILD_ARGS=""
+INTERACTIVE_SHELL=false
+RUN_TESTS=false
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# Logging functions
+log_info() {
+    echo -e "${BLUE}[INFO]${NC} $1"
+}
+
+log_success() {
+    echo -e "${GREEN}[SUCCESS]${NC} $1"
+}
+
+log_warning() {
+    echo -e "${YELLOW}[WARNING]${NC} $1"
+}
+
+log_error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+}
+
+# Help function
+usage() {
+    cat << EOF
+Docker-based build script for StarRocks
+
+Usage: $0 [OPTIONS]
+
+OPTIONS:
+    --fe                    Build Frontend only
+    --be                    Build Backend only
+    --spark-dpp             Build Spark DPP application
+    --hive-udf              Build Hive UDF
+    --clean                 Clean before building
+    --shell                 Open interactive shell in dev environment
+    --test                  Run tests after building
+    --image IMAGE           Use specific Docker image (default: $DOCKER_IMAGE)
+    --help, -h              Show this help message
+
+BUILD OPTIONS (passed to build.sh):
+    --enable-shared-data    Build Backend with shared-data feature support
+    --with-gcov             Build Backend with gcov
+    --without-gcov          Build Backend without gcov (default)
+    --with-bench            Build Backend with bench
+    --with-clang-tidy       Build Backend with clang-tidy
+    --without-java-ext      Build Backend without java-extensions
+    --without-starcache     Build Backend without starcache library
+    --without-tenann        Build without vector index tenann library
+    --without-avx2          Build Backend without avx2 instruction
+    --disable-java-check-style  Disable Java checkstyle checks
+    -j N                    Build with N parallel jobs
+
+EXAMPLES:
+    $0                                          # Build all components
+    $0 --fe --clean                             # Clean and build Frontend
+    $0 --be --with-gcov                         # Build Backend with gcov
+    $0 --shell                                  # Open interactive shell
+    $0 --test                                   # Build and run tests
+    $0 --image starrocks/dev-env-ubuntu:latest  # Use different image
+
+ENVIRONMENT VARIABLES:
+    STARROCKS_DEV_ENV_IMAGE    Docker image to use (default: $DOCKER_IMAGE)
+    DOCKER_BUILD_OPTS          Additional Docker run options
+
+EOF
+}
+
+# Parse command line arguments
+parse_args() {
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            --fe|--be|--spark-dpp|--hive-udf|--clean|--enable-shared-data|--use-staros)
+                BUILD_ARGS="$BUILD_ARGS $1"
+                shift
+                ;;
+            --with-gcov|--without-gcov|--with-bench|--with-clang-tidy|--without-java-ext)
+                BUILD_ARGS="$BUILD_ARGS $1"
+                shift
+                ;;
+            --without-starcache|--without-tenann|--without-avx2|--disable-java-check-style)
```

Comment:
it's a little bit hard to extend when more options added to build.sh or run-fe-ut.sh / run-be-ut.sh